### PR TITLE
fix: navbar dropdown/menu items

### PR DESCRIPTION
**Pull request by plunkettgoogle**
## Description

The "Log Out" option in the dropdown menu in desktop mode didn't extend all the way to the end of the dropdown menu, and the icons on the menu item in mobile mode were getting pushed to the next line.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/addresses technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

### Desktop before:
localhost_3000_account_edit (13)

### Desktop after:
localhost_3000_account_edit (12)

### Mobile before:
localhost_3000_account_edit (14)

### Mobile after:
localhost_3000_account_edit (11)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes
- [ ] I have run `yarn generate:client` if I made backend changes
- [ ] I have exported any new pieces in ui-components

### CODE REVIEW

File: ui-components/src/headers/SiteHeader.scss [hunk 1]
```diff
@@ -281,32 +281,34 @@
         .navbar-dropdown-container {
           background-color: white;
           position: absolute;
-          margin-left: -124px;
+          margin-left: -144px;
```
Comment (GabeBenjamin) [inline ui-components/src/headers/SiteHeader.scss:284]: I think we usually try not to edit ui-components unless this is something thats broken upstream as well and we contribute it back upstream. Is this broken for them or only for us?

If its only for us we should override their css in override.scss

File: ui-components/src/headers/SiteHeader.tsx [hunk 2]
```diff
@@ -113,7 +113,7 @@ const SiteHeader = (props: SiteHeaderProps) => {
 
     return options.map((option, index) => {
       return (
-        <div key={index}>
+        <div className="navbar-dropdown-item-container" key={index}>
```
Comment (GabeBenjamin) [inline ui-components/src/headers/SiteHeader.tsx:116]: Similarly, is this a change that upstream needs? If not, maybe make this more general by taking in a className prop and then have us set our own class in public/

**Review by GabeBenjamin — CHANGES_REQUESTED** [review]

File: ui-components/src/headers/SiteHeader.scss (hunk 1 above)
Comment (plunkettgoogle) [reply ui-components/src/headers/SiteHeader.scss]: I'm not sure if it's broken upstream, as I can't seem to create an account on their dev site. I moved all this stuff into override.scss anyways.

File: ui-components/src/headers/SiteHeader.scss [hunk 3]
```diff
@@ -281,32 +281,34 @@
...
-              &:hover,
-              &:focus {
-                @apply bg-gray-400;
+            .navbar-dropdown-item-container {
+              display: flex;
+              .navbar-dropdown-item {
+                @apply flex;
+                @apply flex-1;
+                @apply items-center;
+                justify-content: left;
```
[the remark this answers is not in the document]
Comment (plunkettgoogle) [reply ui-components/src/headers/SiteHeader.scss]: Done.

File: ui-components/src/headers/SiteHeader.scss [hunk 4]
```diff
@@ -281,32 +281,34 @@
...
+                @apply text-gray-750;
+                @apply pt-3;
+                @apply pb-3;
+                @apply pl-3;
+                @apply pr-3;
+                @apply border-t;
+                @apply border-gray-450;
+                @apply normal-case;
+                font-size: 14px;
```
[the remark this answers is not in the document]
Comment (plunkettgoogle) [reply ui-components/src/headers/SiteHeader.scss]: Done.

File: ui-components/src/headers/SiteHeader.tsx (hunk 2 above)
Comment (plunkettgoogle) [reply ui-components/src/headers/SiteHeader.tsx:116]: I think it's OK to hardcode this className here, as it's done similarly elsewhere in this file (`navbar-dropdown-*`).

File: sites/public/styles/overrides.scss [hunk 5]
```diff
@@ -58,7 +58,9 @@
 
         .navbar-mobile-menu-button {
           border: none;
-          margin: 2.75rem 0.5rem 0.75rem;
+          margin: 2.75rem 0.5rem 0;
+          padding: 0 0.5rem 0.75rem;
+          @apply flex;
```
[the remark this answers is not in the document]
Comment (plunkettgoogle) [reply sites/public/styles/overrides.scss]: Done.

**Comment by plunkettgoogle** [comment]
@GabeBenjamin, how is this looking now?

File: sites/public/styles/overrides.scss [hunk 6]
```diff
@@ -56,9 +56,41 @@
           border: none;
         }
 
+        .navbar-dropdown-container {
+          margin-left: -144px;
```
Comment (GabeBenjamin) [inline sites/public/styles/overrides.scss:60]: Why do we need a -144px margin here? What are we counteracting?

File: sites/public/styles/overrides.scss [hunk 7]
```diff
@@ -56,9 +56,41 @@
...
+          width: auto;
+          .navbar-dropdown {
+            .navbar-dropdown-item-container {
+              display: flex;
+            }
+            .navbar-dropdown-item {
+              @apply border-t;
```
Comment (GabeBenjamin) [inline sites/public/styles/overrides.scss:67]: I believe you can do 1 @apply and add all the styles to the one apply like:
`@apply border-t border-gray-450 flex flex-1 items-center normal-case`

File: sites/public/styles/overrides.scss [hunk 8]
```diff
@@ -56,9 +56,41 @@
...
+              @apply border-gray-450;
+              @apply flex;
+              @apply flex-1;
+              @apply items-center;
+              @apply normal-case;
+              @apply pt-3;
```
Comment (GabeBenjamin) [inline sites/public/styles/overrides.scss:73]: I think all these p's can be combined into `p-3`. What's here is the equivalent of:
```
padding-top: 0.75rem;
padding-bottom: 0.75rem;
padding-left: 0.75rem;
padding-right: 0.75rem;
```
and p-3 is
```
padding: 0.75rem;
```

File: sites/public/styles/overrides.scss [hunk 9]
```diff
@@ -56,9 +56,41 @@
...
+              @apply pb-3;
+              @apply pl-3;
+              @apply pr-3;
+              @apply text-gray-750;
+              font-size: 0.7rem;
```
Comment (GabeBenjamin) [inline sites/public/styles/overrides.scss:78]: Are you sure we want .7rem here? 16 * .7 = 11.2. It's a little weird to do default fractional font sizes. Maybe do .75 (12px)?

File: ui-components/src/headers/SiteHeader.tsx (hunk 2 above)
Comment (GabeBenjamin) [reply ui-components/src/headers/SiteHeader.tsx:116]: Ok, we will just need to upstream this change as well.

**Review by GabeBenjamin — CHANGES_REQUESTED** [review]

File: sites/public/styles/overrides.scss (hunk 6 above)
Comment (plunkettgoogle) [reply sites/public/styles/overrides.scss]: This was basically pulled from SiteHeader.scss. I've kind of reorganized/simplify this stuff, PTAL.

File: sites/public/styles/overrides.scss (hunk 8 above)
Comment (plunkettgoogle) [reply sites/public/styles/overrides.scss]: Done.

File: sites/public/styles/overrides.scss (hunk 7 above)
Comment (plunkettgoogle) [reply sites/public/styles/overrides.scss]: Done.

File: sites/public/styles/overrides.scss (hunk 9 above)
Comment (plunkettgoogle) [reply sites/public/styles/overrides.scss]: Done.

File: ui-components/src/headers/SiteHeader.tsx (hunk 2 above)
Comment (plunkettgoogle) [reply ui-components/src/headers/SiteHeader.tsx:116]: I split this change out into a separate commit to make upstreaming easer.

**Review by GabeBenjamin — APPROVED** [review]

**Pull request merged**